react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,24 @@
+import { FormEvent, useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  function handleSubmit(e: FormEvent<HTMLFormElement>): void {
+    e.preventDefault();
+    console.log(`controlled: username ${username}, password ${password}`);
+  }
+  return (
+    <form onSubmit={handleSubmit}>
+      <h2>Controlled:</h2>
+      <label>
+        <p>Username</p>
+        <input onChange={(e) => setUsername(e.target.value)} />
```

Comment:
Controlled elements need both `value` and `onChange`. Since your inputs start out empty, they work fine, but in a more complex application you'll need to control the value, too.